src/routes/users/fetch_user.rs

```diff
@@ -16,5 +16,15 @@ pub async fn req(user: User, target: Ref) -> Result<Value> {
         Err(Error::MissingPermission)?
     }
 
+    // If the user's status is `Presence::Invisible`, return it as `Presence::Offline`
+    if let Some(mut status) = target.status {
+        if let Some(presence) = status.presence {
+            if presence == Presence::Invisible {
+                status.presence = Some(Presence::Offline);
+                target.status = Some(status);
+            }
+        }
+    }
```

Comment:
This code should be moved to user.rs#L146 (inside fn with) so that this change applies to anywhere that user objects are given to other users.

Also do we need the `mut`?

Reply:
> Also do we need the mut?

Not really. I just find dealing with a mutable value easier.